### DISCUSSION TITLE
Code search does not return results for exact string matches that exist in the repo being searched

### DISCUSSION THREAD

**Discussion by jakeonrails** [community/community#20633] [Other Feature Feedback, Questions, & Ideas]
I searched a private repo on Github, and got 0 results for code. In Visual Studio Code however, I searched the exact same string and got 28 results in 8 files. This so common that Github search is basically useless to me

This issue is a similar one: https://github.com/github-community/community/discussions/18624

**Comment by look** [comment]
Are you using the Code Search Technology Preview?

**Comment by smellai** [comment]
I wrote a node script to search for NPM packages installed in my org repositories, I'm doing this using GitHub code search through **octokit**:
```
GET /search/code
```
with
```
q: `"${pkgName}" in:file org:${org} filename:package.json`
```


If the pkgName has a scope with a hyphen character in it, the search returns zero results. If the hyphen character is not in the scope prefix, but in the package name itself, the search works fine. You can see this behaviour happening using the GitHub search through GitHub interface too:

[search for "typescript-eslint/parser" in mui org](https://github.com/search?q=org%3Amui+%22typescript-eslint%2Fparser%22&type=code) : zero results. But you can see that this [material-ui file](https://github.com/mui/material-ui/blob/master/package.json#L109) contains the string.

[search for "@types/format-util" in mui org](https://github.com/search?q=org%3Amui+%22%40types%2Fformat-util%22&type=code): 5 results.

Why is that? I requested access to the cs preview, but I guess there's no way to access its feature through octokit or APIs, right?

**Reply by look** [reply]
@smellai the old code search does not index hyphens, which is why your search does not work. The new code search does index all characters, so it can do exact match searches, but doesn't have an API yet, sorry. It's on our roadmap.

**Reply by smellai** [reply]
it looks like it works fine now, I guess octokit is now using the new codesearch, cool!

**Comment by mennowitteveen** [comment]
I observed similar. I did the following search: 

https://github.com/scikit-learn/scikit-learn/search?q=numpy

<img width="900" alt="image" src="https://user-images.githubusercontent.com/6292714/231415931-d3503ae4-e299-4d4e-9865-150b769bd483.png">

There are many "numpy" strings in the scikit-learn repo.

**Reply by look** [reply]
Sorry about that. Please try the new search: https://github.com/features/code-search

**Reply by arakji** [reply]
Search for prvCreateIdleTasks in FreeRtos shows zero results. But it is tasks.c.

**Reply by look** [reply]
@arakji sorry for the inconvenience. This is because `tasks.c` is too large to be indexed as [documented in the limitations](https://docs.github.com/en/search-github/github-code-search/about-github-code-search#limitations).
